src/Platform/Microsoft.Testing.Platform/Hosts/TestHostControllersTestHost.cs

```diff
@@ -211,7 +211,15 @@ protected override async Task<int> InternalRunAsync()
                 _logger.LogDebug($"Test host process exited, PID: '{processExited?.Id}'");
             };
 
-            await _logger.LogDebugAsync($"Started test host process '{testHostProcess.Id}' HasExited: {testHostProcess.HasExited}");
+            try
+            {
+                await _logger.LogDebugAsync($"Started test host process '{testHostProcess.Id}' HasExited: {testHostProcess.HasExited}");
+            }
+            catch (InvalidOperationException)
+            {
+                // Access PID can throw InvalidOperationException if the process has already exited:
+                // System.InvalidOperationException: No process is associated with this object.
+            }
```

Comment:
we could be more precise and check here if exited otherwise throw...we could mask something maybe

https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.hasexited?view=net-8.0